File: crossecore-java/src/java_/AbstractMethodDeclaration.java
package java_;
public interface AbstractMethodDeclaration 
 extends BodyDeclaration
{
	
	public static com.crossecore.ocl.QuickSet<AbstractMethodDeclaration> allInstances_ = new com.crossecore.ocl.QuickSet<AbstractMethodDeclaration>(AbstractMethodDeclaration.class);
	
		
	public static com.crossecore.ocl.QuickSet<AbstractMethodDeclaration> allInstances(){
		
		com.crossecore.ocl.QuickSet<AbstractMethodDeclaration> result = new com.crossecore.ocl.QuickSet<AbstractMethodDeclaration>(AbstractMethodDeclaration.class);
		result.addAll(AbstractMethodDeclaration.allInstances_);
		
		result.addAll(MethodDeclaration.allInstances_);
		result.addAll(UnresolvedMethodDeclaration.allInstances_);
		result.addAll(ConstructorDeclaration.allInstances_);
		
		return result;
	}
	
	//public static com.crossecore.ocl.QuickSet<AbstractMethodDeclaration> allInstances = new com.crossecore.ocl.QuickSet<AbstractMethodDeclaration>(AbstractMethodDeclaration.class);
	
	
	Block getBody();
	void setBody(Block value);
	com.crossecore.ocl.OrderedSet<SingleVariableDeclaration> getParameters();
	com.crossecore.ocl.OrderedSet<TypeAccess> getThrownExceptions();
	com.crossecore.ocl.OrderedSet<TypeParameter> getTypeParameters();
	com.crossecore.ocl.Set<MethodRef> getUsagesInDocComments();
	com.crossecore.ocl.Set<AbstractMethodInvocation> getUsages();
	
}

File: crossecore-typescript/java_/AbstractMethodDeclaration.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {AbstractMethodInvocation} from "./AbstractMethodInvocation";
import {TypeParameter} from "./TypeParameter";
import {MethodRef} from "./MethodRef";
import {SingleVariableDeclaration} from "./SingleVariableDeclaration";
import {Block} from "./Block";
import {TypeAccess} from "./TypeAccess";
import {BodyDeclaration} from "./BodyDeclaration";

export interface AbstractMethodDeclaration
extends BodyDeclaration

{
	
	body:Block;
	parameters: OrderedSet<SingleVariableDeclaration>;
	thrownExceptions: OrderedSet<TypeAccess>;
	typeParameters: OrderedSet<TypeParameter>;
	usagesInDocComments: Set<MethodRef>;
	usages: Set<AbstractMethodInvocation>;
	

}

